Rename Image::hasSingleSecurityOrigin to Image::renderingTaintsOrigin.

mattwoodrow opened the pull request:
#### 5cfdacb9f77cffe585529fa7a25afc82234403c9
<pre>
Rename Image::hasSingleSecurityOrigin to Image::renderingTaintsOrigin.
<a href="https://bugs.webkit.org/show_bug.cgi?id=250540">https://bugs.webkit.org/show_bug.cgi?id=250540</a>
&lt;rdar://102738351&gt;

Reviewed by Darin Adler.

There&apos;s only two entry points to this function, and one of them (CanvasRenderingContext) only calls it when the image is a SVGImage.

The only interesting implementation of this (SVGImage) is not related to multiple origins (or CORS at all), but instead to SVG foreign object possibly tainting the &lt;canvas&gt; in a different way (by drawing visited colours, spellcheck etc).

This renames the function to renderingTaintsOrigin, since that more accurately reflects the behaviour. It also swaps the meaning in the process, and removes lots of overloads which aren&apos;t needed.

The only (potential) behaviour change is that the ImageBitmap callsite (which didn&apos;t have an isSVGImage check) will now definitely only affect SVGImages instead of other types (which implicitly returned false for hasSingleSecurity origin by not overloading the function). I believe that was the intent of the change which added the check to the other callsite (and that returning false for !SVG was a missing impl bug), and this was just missed.

* Source/WebCore/html/ImageBitmap.cpp:
(WebCore::taintsOrigin):
* Source/WebCore/html/canvas/CanvasRenderingContext.cpp:
(WebCore::CanvasRenderingContext::taintsOrigin):
(WebCore::CanvasRenderingContext::checkOrigin):
(WebCore::CanvasRenderingContext::wouldTaintOrigin): Deleted.
* Source/WebCore/html/canvas/CanvasRenderingContext.h:
(WebCore::CanvasRenderingContext::checkOrigin):
* Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp:
(WebCore::WebGLRenderingContextBase::validateHTMLImageElement):
(WebCore::WebGLRenderingContextBase::validateHTMLCanvasElement):
(WebCore::WebGLRenderingContextBase::validateHTMLVideoElement):
* Source/WebCore/platform/graphics/BitmapImage.h:
* Source/WebCore/platform/graphics/GeneratedImage.h:
* Source/WebCore/platform/graphics/Image.h:
(WebCore::Image::renderingTaintsOrigin const):
(WebCore::Image::hasSingleSecurityOrigin const): Deleted.
* Source/WebCore/platform/graphics/cg/PDFDocumentImage.h:
* Source/WebCore/svg/SVGFEImageElement.cpp:
(WebCore::SVGFEImageElement::renderingTaintsOrigin const):
(WebCore::SVGFEImageElement::hasSingleSecurityOrigin const): Deleted.
* Source/WebCore/svg/SVGFEImageElement.h:
* Source/WebCore/svg/SVGImageElement.cpp:
(WebCore::SVGImageElement::renderingTaintsOrigin const):
(WebCore::SVGImageElement::hasSingleSecurityOrigin const): Deleted.
* Source/WebCore/svg/SVGImageElement.h:
* Source/WebCore/svg/graphics/SVGImage.cpp:
(WebCore::SVGImage::renderingTaintsOrigin const):
(WebCore::SVGImage::hasSingleSecurityOrigin const): Deleted.
* Source/WebCore/svg/graphics/SVGImage.h:

Canonical link: <a href="https://commits.webkit.org/258969@main">https://commits.webkit.org/258969@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/42ac29d13933f1fc00922e6c942211755f82e7d2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/103495 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/12613 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/36456 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/112731 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/172941 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/107449 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/13645 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/3517 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/95751 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/111912 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/109269 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/10500 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/93576 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/38241 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/92321 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/25162 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/79885 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/6000 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/26565 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/6177 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/3096 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/12159 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/46074 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6161 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/7932 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

darinadler reviewed (approved):
This is an improvement, but the naming is still not super clear.

Perhaps functions that don’t answer the entire question should be named to express the part of the question they do an do not answer. Like why is the `protocolIsData` check in the `CanvasRenderingContext::wouldTaintOrigin` function rather than in the `Image::wouldTaintOrigin` function? Seems risky to have a function in `Image` that does part but not all of the check with the **same name** as the definitive function you should call to do the entire correct check.

I don’t understand why the function inside `ImageBitmap.cpp` is called `taintsOrigin`, but the others are called `wouldTaintOrigin`. Maybe the function in `ImageBitmap.cpp` should be renamed to match.

mattwoodrow committed:
Rename Image::hasSingleSecurityOrigin to Image::renderingTaintsOrigin.
https://bugs.webkit.org/show_bug.cgi?id=250540
<rdar://102738351>

Reviewed by Darin Adler.

There's only two entry points to this function, and one of them (CanvasRenderingContext) only calls it when the image is a SVGImage.

The only interesting implementation of this (SVGImage) is not related to multiple origins (or CORS at all), but instead to SVG foreign object possibly tainting the <canvas> in a different way (by drawing visited colours, spellcheck etc).

This renames the function to renderingTaintsOrigin, since that more accurately reflects the behaviour. It also swaps the meaning in the process, and removes lots of overloads which aren't needed.

The only (potential) behaviour change is that the ImageBitmap callsite (which didn't have an isSVGImage check) will now definitely only affect SVGImages instead of other types (which implicitly returned false for hasSingleSecurity origin by not overloading the function). I believe that was the intent of the change which added the check to the other callsite (and that returning false for !SVG was a missing impl bug), and this was just missed.

* Source/WebCore/html/ImageBitmap.cpp:
(WebCore::taintsOrigin):
* Source/WebCore/html/canvas/CanvasRenderingContext.cpp:
(WebCore::CanvasRenderingContext::taintsOrigin):
(WebCore::CanvasRenderingContext::checkOrigin):
(WebCore::CanvasRenderingContext::wouldTaintOrigin): Deleted.
* Source/WebCore/html/canvas/CanvasRenderingContext.h:
(WebCore::CanvasRenderingContext::checkOrigin):
* Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp:
(WebCore::WebGLRenderingContextBase::validateHTMLImageElement):
(WebCore::WebGLRenderingContextBase::validateHTMLCanvasElement):
(WebCore::WebGLRenderingContextBase::validateHTMLVideoElement):
* Source/WebCore/platform/graphics/BitmapImage.h:
* Source/WebCore/platform/graphics/GeneratedImage.h:
* Source/WebCore/platform/graphics/Image.h:
(WebCore::Image::renderingTaintsOrigin const):
(WebCore::Image::hasSingleSecurityOrigin const): Deleted.
* Source/WebCore/platform/graphics/cg/PDFDocumentImage.h:
* Source/WebCore/svg/SVGFEImageElement.cpp:
(WebCore::SVGFEImageElement::renderingTaintsOrigin const):
(WebCore::SVGFEImageElement::hasSingleSecurityOrigin const): Deleted.
* Source/WebCore/svg/SVGFEImageElement.h:
* Source/WebCore/svg/SVGImageElement.cpp:
(WebCore::SVGImageElement::renderingTaintsOrigin const):
(WebCore::SVGImageElement::hasSingleSecurityOrigin const): Deleted.
* Source/WebCore/svg/SVGImageElement.h:
* Source/WebCore/svg/graphics/SVGImage.cpp:
(WebCore::SVGImage::renderingTaintsOrigin const):
(WebCore::SVGImage::hasSingleSecurityOrigin const): Deleted.
* Source/WebCore/svg/graphics/SVGImage.h:

Canonical link: https://commits.webkit.org/258969@main
